modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -163,7 +163,9 @@ export default {
     finishExit () {
       this.removeEventListeners();
       this.$emit('no-modal');
-      apos.modal.stack.pop();
+      // pop doesn't quite suffice because of race conditions when
+      // closing one and opening another
```

Comment:
Fixes a bug that breaks the escape key if modals are opened while others are technically still closing.